pkg/node/chain_stub.go

```diff
@@ -0,0 +1,128 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/postage/postagecontract"
+	"github.com/ethersphere/bee/pkg/settlement/swap/chequebook"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+type stubMatcher struct{}
+
+func (m *stubMatcher) Matches(context.Context, []byte, uint64, swarm.Address, bool) ([]byte, error) {
+	return nil, nil
+}
+
+func stubSwapBackend(log logging.Logger) *loggingSwapBackend {
+	return &loggingSwapBackend{
+		log: log,
+	}
+}
+
+type loggingSwapBackend struct {
+	log logging.Logger
+}
+
+func (m loggingSwapBackend) CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error) {
+	m.log.Debug("MockSwapBackend: CodeAt")
+	return common.FromHex(sw3abi.SimpleSwapFactoryDeployedBinv0_4_0), nil
+}
+func (m loggingSwapBackend) CallContract(ctx context.Context, call ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	panic("MockSwapBackend: CallContract")
+}
+func (m loggingSwapBackend) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
+	m.log.Debug("MockSwapBackend: HeaderByNumber")
+	h := new(types.Header)
+	h.Time = uint64(time.Now().Unix())
+	return h, nil
+}
+func (m loggingSwapBackend) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	panic("MockSwapBackend: PendingNonceAt")
```

Comment:
These methods panic because they are not expected to be called.
If they still get called it means it's a programmatic error and it makes a valid case for panics.